src/net_processing.cpp

```diff
@@ -556,8 +515,44 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
         // Potentially mark this peer as a preferred download peer.
         UpdatePreferredDownload(pfrom);
 
-        // Send VERACK handshake message
-        pfrom->PushMessage(NetMsgType::VERACK);
+        if (pfrom->nServices & NODE_XVERSION)
+        {
+            // BU expedited procecessing requires the exchange of the listening port id
+            // The former BUVERSION message has now been integrated into the xmap field in CXVersionMessage.
+
+            // prepare xversion message. This *must* be the next message after the verack has been received,
+            // if it comes at all.
```

Comment:
the comment says that the xversion must be the next message "after" the verack...i thought you were now sending the verack after the xversion?

Reply:
good find, i will fix that comment.

Reply:
fixed